doc/auto_techsupport/auto_techsupport_and_coredump_mgmt.md

```diff
@@ -458,12 +460,45 @@ Although if the admin feels otherwise, these values are configurable.
+Recently, an enhancement was made to techsupport script to only run one instance at a time by using a locking mechanism. When other instance of techsupport tries to run, it'll exit with a relevent code. This would apply nevertheless of how a techsupport was invoked i.e. manual or through auto-techsupport. 
 
 With this change, rate-limit-interval of zero would not make any difference. The locking mechanism would implicitly impose a minimum rate-limit-interval of techsupport execution time.  And since, the techsupport execution time can't be found out and varies based on underlying machine and system state, the range of values configurable for the rate-limit-interval is left unchanged
 
 A relevant message will be logged to syslog when the invocation fails because of LOCKFAIL exit code. 
 
+### 7.9 Orchagent abort consideration 
+
+Note: This solution currently applies only for Mellanox platforms.
+
+When the orchagent deems a SAI CRUD call as a failure based on the return status, it aborts itself. 
+
+This'll result in 
+1. A core dump to be generated and it triggers auto-techsupport if enabled
+2. All the services are restarted including syncd 
+
+So, it is highly likely that by the time auto-techsupport collects saisdkdump, syncd might have been restarted or in the process of restarting. In either case, we'd be loosing the saisdkdump information before restart which will contain useful information for triaging. Thus, a special handling is needed for the core dumps generated from swss container.
+
+This requires enhancements not just in auto-techsupport but also in orchagent process and also in /usr/local/bin/syncd.sh script. 
+
+![Orch Abort Flow](orch_abort_flow.PNG)
+
+Firstly, orchagent can terminate because of various reasons, i.e. some failure in orchagent like SEGFAULT, config reload etc but we are only interested in the case of SAI programming failure. During a SAI programming failure, orchagent will call abort i.e. kernel will send SIGABRT and thus the process terminates. To differentiate among these cases, orchagent will write to a table in STATE_DB named "ORCH_ABRT_STATUS".
+
+#### Schema: 
+```
+ORCH_ABRT_STATUS
+<1|empty>
+
+Eg:
+root@sonic:/home/admin# sonic-db-cli STATE_DB GET ORCH_ABRT_STATUS
+1
+```
+
+During sai programming failure, orchagent will set the status to ORCH_ABRT_STATUS flag in STATE_DB. syncd.sh script checks if the ORCH_ABRT_STATUS flag is set in STATE_DB before stopping the syncd container and if yes proceeds with collecting saisdkdump to `/var/log/orch_abrt_saisdkdump/` on the host and also creates a file under /tmp named 'saidump_collection_notify_flag'. This is used to synchronize b/w auto-techsupport and syncd.

```

Comment:
assuming it will be generic and some ASIC vendors will nor refer to the new state db adds, what will be the system behaviour?

Reply:
Orchagent will write to STATE_DB irrespective of the vendor. In the syncd.sh scrip
    if [[ x"$(${SONIC_DB_CLI} STATE_DB GET ORCH_ABRT_STATUS)" == x"1" ]]; then
        # Collecting saisdkdump before restarting syncd
        # Runs when orchagent is aborted because of SAI failure.
        # Only enabled for mellanox platform
        if [[ x$sonic_asic_platform == x"mellanox" ]]; then
            collect_saisdkdump
        fi
        # This is used to notify auto-techsupport process
        touch /tmp/saidump_collection_notify_flag
    fi